code/game/objects/items/weapons/holy_weapons.dm

```diff
@@ -0,0 +1,328 @@
+/obj/item/weapon/nullrod
+	name = "null rod"
+	desc = "A rod of pure obsidian, its very presence disrupts and dampens the powers of Nar-Sie's followers."
+	icon_state = "nullrod"
+	item_state = "nullrod"
+	force = 18
+	throw_speed = 3
+	throw_range = 4
+	throwforce = 10
+	w_class = 1
+	var/reskinned = FALSE
+	var/list/fluff_transformations = list() //does it have any special transformations only accessible to it? Should only be subtypes of /obj/item/weapon/nullrod
+
+/obj/item/weapon/nullrod/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] is killing \himself with \the [src.name]! It looks like \he's trying to get closer to god!</span>")
+	return (BRUTELOSS|FIRELOSS)
+
+/obj/item/weapon/nullrod/attack(mob/M, mob/living/carbon/user)
+	..()
+	if(M.mind)
+		if(M.mind.vampire)
+			if(ishuman(M))
+				if(!M.mind.vampire.get_ability(/datum/vampire_passive/full))
+					to_chat(M, "<span class='warning'>The nullrod's power interferes with your own!</span>")
+					M.mind.vampire.nullified = max(5, M.mind.vampire.nullified + 2)
+
+/obj/item/weapon/nullrod/attack_self(mob/user)
+	if(reskinned)
+		return
+	if(user.mind && (user.mind.assigned_role == "Chaplain"))
+		reskin_holy_weapon(user)
+
+/obj/item/weapon/nullrod/proc/reskin_holy_weapon(mob/M)
+	var/list/holy_weapons_list = typesof(/obj/item/weapon/nullrod) - typesof(/obj/item/weapon/nullrod/fluff)
+	if(fluff_transformations.len)
+		for(var/thing in fluff_transformations)
+			holy_weapons_list += thing
+	var/list/display_names = list()
+	for(var/V in holy_weapons_list)
+		var/atom/A = V
+		display_names += initial(A.name)
+
+	var/choice = input(M,"What theme would you like for your holy weapon?","Holy Weapon Theme") as null|anything in display_names
+	if(!src || !choice || M.stat || !in_range(M, src) || M.restrained() || !M.canmove || reskinned)
```

Comment:
`M.stat`, `M.restrained()`, and `!M.canmove` need to be replaced by `M.incapacitated()`